zetaclient/chains/evm/observer/v2_inbound.go

```diff
@@ -149,7 +151,58 @@ func (ob *Observer) parseAndValidateDepositEvents(
 		return validEvents[i].Raw.BlockNumber < validEvents[j].Raw.BlockNumber
 	})
 
-	return validEvents
+	// Check if multiple calls are enabled
+	if ob.shouldAllowMultipleCalls(ctx) {
+		return validEvents
+	}
+	return ob.filterDepositEventsByTx(validEvents)
+}
+
+// shouldAllowMultipleCalls checks the EnableMultipleCalls feature flag
+func (ob *Observer) shouldAllowMultipleCalls(ctx context.Context) bool {
+	// Get app context to access feature flags
+	app, err := zctx.FromContext(ctx)
+	if err != nil {
+		ob.Logger().Inbound.Warn().Err(err).Msg("unable to get app context, using default behavior")
+		// Fallback to filtering (return false)
+		return false
+	}
+
+	// Check feature flag to determine whether to allow multiple calls from same tx
+	cfg := app.Config()
+
+	// If EnableMultipleCalls is enabled, return all valid events without filtering
+	if cfg.IsEnableMultipleCallsEnabled() {
+		ob.Logger().Inbound.Debug().
+			Msg("EnableMultipleCalls enabled: returning all valid events without filtering")
+		return true
+	}
+
+	// Default behavior: filter events from same tx (only first event)
+	ob.Logger().Inbound.Debug().
+		Msg("EnableMultipleCalls disabled: filtering multiple events from same tx (only first event)")
+	return false
+}
+
+// filterDepositEventsByTx filters deposit events from same tx (original logic)
```

Comment:
We will remove this option at some point once it's battle-tested (will be the same for Solana), we should open an issue and add link for it here as a TODO to remove it